### PR TITLE
updated ExecutingASimulatorTest page for simulator doc.

### PR DESCRIPTION
**Pull request by hasancelik**

### CODE REVIEW

**Comment by pveentjer** [comment]
The scope of the page is not how to run any test, but one of the predefined Hazelcast tests e.g. StringStringMapTest. So there should not be anything regarding classpaths.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 1]
```diff
@@ -4,17 +4,311 @@
 
 After you install and prepare the Hazelcast Simulator for your environment, it is time to perform a test.
 
-The following steps execute a Hazelcast Simulator test.
+In the following sections, we provide an example test and its output along with the required edits to the files `simulator.properties` and `test.properties`.
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: We don't provide an test; the user is going to run an existing test.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 2]
```diff
@@ -4,17 +4,311 @@
...
-4. Edit the `simulator.properties` file according to your needs.
-5. Copy the `test.properties` file from the `/simulator-tests` folder of Hazelcast Simulator to your working folder.
-6. Edit the `test.properties` file according to your needs.
-5. Execute the `run.sh` script while you are in your working folder to perform your Simulator test.
+### Creating and editing properties file
 
-In the following sections, we provide an example test and its output along with the required edits to the files `simulator.properties` and `test.properties`.
+***NOTE:*** *For a full description of the file `simulator.properties`, please see the [Simulator.Properties File Description section](#simulator-properties-file-description). Also you can find sample simulator properties in the `dist/simulator-tests/simulator.properties`. You can copy the this file to working directory ,then you can edit according to your needs.*
+
+You need to give the classpath of `test` in the file `test.properties` as shown below.
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: no reason to task about class path.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 3]
```diff
@@ -4,17 +4,311 @@
...
+
+```
+example@class=com.hazelcast.simulator.tests.ExampleTest
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: there does not exist a example test. Use StringStringMapTest.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 4]
```diff
@@ -4,17 +4,311 @@
...
+example@maxKeys=5000
+example@putProb=0.4
+```
+
+* The `example` word at the start of the each line, represents `id` of the `test suite`. 
+* The property `class` defines the actual test case and the rest are the properties you want to bind in your test. If a
+property is not defined in this file, the default value of the property given in your test code is used. Please see the `properties` comment in the example code above.
+
+You can also define multiple tests in the file `test.properties` as shown below.
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Not relevant. Noise for the time being.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 5]
```diff
@@ -4,17 +4,311 @@
...
+INFO  13:27:50 Starting Agent on: 54.157.2.215
+INFO  13:27:51     54.157.2.215 SIMULATOR AGENT STARTED
+INFO  13:27:51 Pausing for machine warmup... (10 sec)
+INFO  13:28:01 Duration: 00d 00h 02m 24s
+INFO  13:28:01 ==============================================================
+INFO  13:28:01 Successfully provisioned 4 aws-ec2 machines
+INFO  13:28:01 ==============================================================
+INFO  13:28:01 Shutting down Provisioner...
+INFO  13:28:01 Done!
+```
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Explain to the end user he can use 'less agents.txt' to see the content of the files.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 6]
```diff
@@ -4,17 +4,311 @@
...
+Also you can see `agents.txt` was created automatically by provisioner in the working directory. This `agents.txt` file includes ips of the EC2 instances.
+
+* After the create instances and install agents to them. You need cordinate and configure your test suite via `cooordinator`.
+	
+	```
+	coordinator     --memberWorkerCount 2 \
+                --workerVmOptions "-ea -server -Xms2G -Xmx2G -verbosegc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -Xloggc:gc.log -XX:+HeapDumpOnOutOfMemoryError" \
+                --hzFile            hazelcast.xml \
+                --clientWorkerCount 2 \
+                --clientWorkerVmOptions "-ea -server -Xms2G -Xmx2G -verbosegc -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -Xloggc:gc.log -XX:+HeapDumpOnOutOfMemoryError" \
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: You don't need such a complex cli.

coordinator test.properties

Is sufficient.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 7]
```diff
@@ -4,17 +4,311 @@
...
-2. Create a folder85 for your tests. Let's call it your working folder.
-3. Copy the `simulator.properties` file from the `/conf` folder of the Hazelcast Simulator to your working folder.
-4. Edit the `simulator.properties` file according to your needs.
-5. Copy the `test.properties` file from the `/simulator-tests` folder of Hazelcast Simulator to your working folder.
-6. Edit the `test.properties` file according to your needs.
-5. Execute the `run.sh` script while you are in your working folder to perform your Simulator test.
+### Creating and editing properties file
 
-In the following sections, we provide an example test and its output along with the required edits to the files `simulator.properties` and `test.properties`.
+***NOTE:*** *For a full description of the file `simulator.properties`, please see the [Simulator.Properties File Description section](#simulator-properties-file-description). Also you can find sample simulator properties in the `dist/simulator-tests/simulator.properties`. You can copy the this file to working directory ,then you can edit according to your needs.*
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Give command to copy this file.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 6 above)
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: There does not exists a file 'client-hazelcast.xml' and 'hazelcast.xml' in the current director. Did you try this at all?

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md [hunk 8]
```diff
@@ -4,17 +4,311 @@
...
+INFO  13:39:54 Duration: 00d 00h 01m 25s
+INFO  13:39:54 ==============================================================
+INFO  13:39:54 Terminated 4 of 4, remaining=0
+INFO  13:39:54 ==============================================================
+INFO  13:39:54 Shutting down Provisioner...
+INFO  13:39:54 Done!
+```
+
+Alternative way to run test via script. The script `run.sh` is for your convenience. It gathers all the commands used to perform a test into one script. The following is the content of this example `run.sh` script.
+
```
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Give the commands to make this script. Including the chmod +x .

**Comment by pveentjer** [comment]
The document in its current state is not suitable as a tutorial how to run a test after you have done a Hz installation. It is full of assumptions. A user needs to execute your instructions step by step; if there is confusion because you assume things; the document is a source of frustration instead of a source of help.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 7 above)
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Edit:

For now we are going to make use of this properties file. And then you give a the command to create that simulator.properties file with the trimmed down content I talked about before. A user should not need to have linux skills to get this stuff done.

**Comment by pveentjer** [comment]
The only way this document is going to be useful is to try every step yourself

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 3 above)
Comment (hasancelik) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: You can find ExampleTest in the simulator repo's test module. Imho ExampleTest more suitable than StringStringMapTest for user use simulator first time. He/She can easily understand what is going on here.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 6 above)
Comment (hasancelik) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: I think we need this cli but you are right it is complex. We can decrease complexity. User should know how to manage member and client from coordinator. We mention each coordinator argument  at another page but execSimTest page is the only practical page in our documentation.

I ran and it is working.

**Comment by hasancelik** [comment]
I performed every step in this doc. It is working properly.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 8 above)
Comment (hasancelik) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: You are right. It is more reasonable.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 6 above)
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: So it is working even though hazelcast.xml is not in the working directory? If this is the case; then there is a bug. If you have hazelcast.xml in your directory; then there is clearly something wrong.

Please use my suggested setttings for coordinator. No need currently to deal with complexity

**Comment by pveentjer** [comment]
Try again. And begin from an empty directory.

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 1 above)
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: Let the user use the following test properties content
```
class=com.hazelcast.simulator.tests.map.StringStringMapTest
threadCount=10
keyLocality=Random
keyLength=300
valueLength=300
keyCount=100000
valueCount=100000
putProb=0.2
basename=map
```

File: hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md (hunk 3 above)
Comment (pveentjer) [inline hazelcast-documentation/src/HazelcastSimulator/ExecutingASimulatorTest.md]: I disagree.

**Pull request closed without merging**